### ISSUE TITLE
[Bug]: fail to get shard leaders from QueryCoord

### ISSUE THREAD

**Issue by cjh6666688888** [milvus-io/milvus#19185]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:milvus-dev:milvus-io-2.1.0-7affd9399-20220906
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2):milvus-sdk-go/v2 v2.1.2-0.20220823032848-8832edc6f96e
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. After the Milvus image upgrade, the original data query is abnormal.
2.This is a known problem. It's happened before.
fail to search on all shard leaders, err=All attempts results:\nattempt #1:fail to get shard leaders from QueryCoord: no replica available\nattempt #2:fail to get shard leaders from QueryCoord: no replica available\nattempt #3:fail to get shard leaders from QueryCoord: no replica available\nattempt #4:fail to get shard leaders from QueryCoord: no replica available\nattempt #5:fail to get shard leaders from QueryCoord: no replica available\nattempt #6:fail to get shard leaders from QueryCoord: no replica available\nattempt #7:fail to get shard leaders from QueryCoord: no replica available\nattempt #8:context deadline exceeded\n

### Steps To Reproduce

```markdown
1.The Milvus image was upgraded to the latest version.
2.Query with the original collection data.
3. Throw an exception when querying:fail to get shard leaders from QueryCoord:
```

### Milvus Log

[2022/09/13 17:39:39.094 +08:00] [DEBUG] [timerecord/time_recorder.go:78] ["Search: search request enqueue (0ms)"]
[2022/09/13 17:39:39.094 +08:00] [DEBUG] [proxy/impl.go:2471] ["Search enqueued"] [traceID=722910e39d795ada] [role=proxy] [msgID=435961008929114342] [timestamp=435961719444471809] [db=] [collection=sl_app_video_deduplication] [partitions="[]"] [dsl=] [len(PlaceholderGroup)=524] [OutputFields="[key]"] [search_params="[{\"key\":\"round_decimal\",\"value\":\"-1\"},{\"key\":\"anns_field\",\"value\":\"data\"},{\"key\":\"topk\",\"value\":\"10\"},{\"key\":\"params\",\"value\":\"{\\\"nprobe\\\":50}\"},{\"key\":\"metric_type\",\"value\":\"IP\"}]"] [travel_timestamp=0] [guarantee_timestamp=2]
[2022/09/13 17:39:39.095 +08:00] [DEBUG] [proxy/task_search.go:217] ["translate output fields"] [msgID=435961008929114342] ["output fields"="[key]"]
[2022/09/13 17:39:39.095 +08:00] [DEBUG] [proxy/task_search.go:238] ["create query plan"] [msgID=435961008929114342] [dsl=] ["anns field"=data] ["query info"="topk:10 metric_type:\"IP\" search_params:\"{\\\"nprobe\\\":50}\" round_decimal:-1 "]
[2022/09/13 17:39:39.095 +08:00] [DEBUG] [proxy/task_search.go:260] [Proxy::searchTask::PreExecute] [msgID=435961008929114342] [plan.OutputFieldIds="[100]"] [plan="vector_anns:<field_id:101 query_info:<topk:10 metric_type:\"IP\" search_params:\"{\\\"nprobe\\\":50}\" round_decimal:-1 > placeholder_tag:\"$0\" > output_field_ids:100 "]
[2022/09/13 17:39:39.095 +08:00] [INFO] [proxy/task_search.go:289] ["search PreExecute done."] [msgID=435961008929114342] [travel_ts=18446744073709551615] [guarantee_ts=435961718133751809] [timeout_ts=0]
[2022/09/13 17:39:39.095 +08:00] [INFO] [proxy/meta_cache.go:559] ["no shard cache for collection, try to get shard leaders from QueryCoord"] [collectionName=sl_app_video_deduplication]
[2022/09/13 17:39:39.096 +08:00] [DEBUG] [retry/retry.go:39] ["retry func failed"] ["retry time"=0] [error="fail to get shard leaders from QueryCoord: no replica available"]
[2022/09/13 17:39:39.300 +08:00] [DEBUG] [proxy/impl.go:2429] ["Search received"] [traceID=e9fec5b50bea35a] [role=proxy] [db=] [collection=search_img_deduplication] [partitions="[]"] [dsl=] [len(PlaceholderGroup)=1548] [OutputFields="[key]"] [search_params="[{\"key\":\"params\",\"value\":\"{\\\"nprobe\\\":50}\"},{\"key\":\"metric_type\",\"value\":\"IP\"},{\"key\":\"round_decimal\",\"value\":\"-1\"},{\"key\":\"anns_field\",\"value\":\"data\"},{\"key\":\"topk\",\"value\":\"10\"}]"] [travel_timestamp=0] [guarantee_timestamp=2]
[2022/09/13 17:39:39.300 +08:00] [DEBUG] [allocator/id.go:140] ["IDAllocator pickCanDoFunc"] [need=1] [total=197722] [remainReqCnt=0]
[2022/09/13 17:39:39.300 +08:00] [DEBUG] [timerecord/time_recorder.go:78] ["Search: search request enqueue (0ms)"]
[2022/09/13 17:39:39.300 +08:00] [DEBUG] [proxy/impl.go:2471] ["Search enqueued"] [traceID=e9fec5b50bea35a] [role=proxy] [msgID=435961008929114343] [timestamp=435961719496900609] [db=] [collection=search_img_deduplication] [partitions="[]"] [dsl=] [len(PlaceholderGroup)=1548] [OutputFields="[key]"] [search_params="[{\"key\":\"params\",\"value\":\"{\\\"nprobe\\\":50}\"},{\"key\":\"metric_type\",\"value\":\"IP\"},{\"key\":\"round_decimal\",\"value\":\"-1\"},{\"key\":\"anns_field\",\"value\":\"data\"},{\"key\":\"topk\",\"value\":\"10\"}]"] [travel_timestamp=0] [guarantee_timestamp=2]
[2022/09/13 17:39:39.300 +08:00] [DEBUG] [proxy/task_search.go:217] ["translate output fields"] [msgID=435961008929114343] ["output fields"="[key]"]
[2022/09/13 17:39:39.300 +08:00] [DEBUG] [proxy/task_search.go:238] ["create query plan"] [msgID=435961008929114343] [dsl=] ["anns field"=data] ["query info"="topk:10 metric_type:\"IP\" search_params:\"{\\\"nprobe\\\":50}\" round_decimal:-1 "]
[2022/09/13 17:39:39.300 +08:00] [DEBUG] [proxy/task_search.go:260] [Proxy::searchTask::PreExecute] [msgID=435961008929114343] [plan.OutputFieldIds="[100]"] [plan="vector_anns:<field_id:101 query_info:<topk:10 metric_type:\"IP\" search_params:\"{\\\"nprobe\\\":50}\" round_decimal:-1 > placeholder_tag:\"$0\" > output_field_ids:100 "]
[2022/09/13 17:39:39.300 +08:00] [INFO] [proxy/task_search.go:289] ["search PreExecute done."] [msgID=435961008929114343] [travel_ts=18446744073709551615] [guarantee_ts=435961718186180609] [timeout_ts=0]
[2022/09/13 17:39:39.301 +08:00] [INFO] [proxy/meta_cache.go:559] ["no shard cache for collection, try to get shard leaders from QueryCoord"] [collectionName=search_img_deduplication]
[2022/09/13 17:39:39.301 +08:00] [DEBUG] [retry/retry.go:39] ["retry func failed"] ["retry time"=0] [error="fail to get shard leaders from QueryCoord: no replica available"]
[2022/09/13 17:39:39.928 +08:00] [DEBUG] [proxy/impl.go:2429] ["Search received"] [traceID=2edd64468627335b] [role=proxy] [db=] [collection=video_deduplication] [partitions="[]"] [dsl=] [len(PlaceholderGroup)=524] [OutputFields="[key]"] [search_params="[{\"key\":\"round_decimal\",\"value\":\"-1\"},{\"key\":\"anns_field\",\"value\":\"data\"},{\"key\":\"topk\",\"value\":\"10\"},{\"key\":\"params\",\"value\":\"{\\\"nprobe\\\":50}\"},{\"key\":\"metric_type\",\"value\":\"IP\"}]"] [travel_timestamp=0] [guarantee_timestamp=2]
[2022/09/13 17:39:39.928 +08:00] [DEBUG] [allocator/id.go:140] ["IDAllocator pickCanDoFunc"] [need=1] [total=197721] [remainReqCnt=0]
[2022/09/13 17:39:39.928 +08:00] [DEBUG] [timerecord/time_recorder.go:78] ["Search: search request enqueue (0ms)"]
[2022/09/13 17:39:39.928 +08:00] [DEBUG] [proxy/impl.go:2471] ["Search enqueued"] [traceID=2edd64468627335b] [role=proxy] [msgID=435961008929114344] [timestamp=435961719667294210] [db=] [collection=video_deduplication] [partitions="[]"] [dsl=] [len(PlaceholderGroup)=524] [OutputFields="[key]"] [search_params="[{\"key\":\"round_decimal\",\"value\":\"-1\"},{\"key\":\"anns_field\",\"value\":\"data\"},{\"key\":\"topk\",\"value\":\"10\"},{\"key\":\"params\",\"value\":\"{\\\"nprobe\\\":50}\"},{\"key\":\"metric_type\",\"value\":\"IP\"}]"] [travel_timestamp=0] [guarantee_timestamp=2]
[2022/09/13 17:39:39.928 +08:00] [DEBUG] [proxy/task_search.go:217] ["translate output fields"] [msgID=435961008929114344] ["output fields"="[key]"]
[2022/09/13 17:39:39.928 +08:00] [DEBUG] [proxy/task_search.go:238] ["create query plan"] [msgID=435961008929114344] [dsl=] ["anns field"=data] ["query info"="topk:10 metric_type:\"IP\" search_params:\"{\\\"nprobe\\\":50}\" round_decimal:-1 "]
[2022/09/13 17:39:39.929 +08:00] [DEBUG] [proxy/task_search.go:260] [Proxy::searchTask::PreExecute] [msgID=435961008929114344] [plan.OutputFieldIds="[100]"] [plan="vector_anns:<field_id:101 query_info:<topk:10 metric_type:\"IP\" search_params:\"{\\\"nprobe\\\":50}\" round_decimal:-1 > placeholder_tag:\"$0\" > output_field_ids:100 "]
[2022/09/13 17:39:39.929 +08:00] [INFO] [proxy/task_search.go:289] ["search PreExecute done."] [msgID=435961008929114344] [travel_ts=18446744073709551615] [guarantee_ts=435961718356574210] [timeout_ts=0]
[2022/09/13 17:39:39.929 +08:00] [INFO] [proxy/meta_cache.go:559] ["no shard cache for collection, try to get shard leaders from QueryCoord"] [collectionName=video_deduplication]
[2022/09/13 17:39:39.930 +08:00] [DEBUG] [retry/retry.go:39] ["retry func failed"] ["retry time"=0] [error="fail to get shard leaders from QueryCoord: no replica available"]
[2022/09/13 17:39:40.401 +08:00] [WARN] [grpclog/grpclog.go:46] ["[core][Server #7] grpc: Server.Serve failed to create ServerTransport: connection error: desc = \"transport: http2Server.HandleStreams failed to receive the preface from client: read tcp 10.244.9.28:19530->172.28.37.18:12972: read: connection reset by peer\""]
[2022/09/13 17:39:40.697 +08:00] [DEBUG] [proxy/impl.go:2429] ["Search received"] [traceID=3f3e3cd14dc83716] [role=proxy] [db=] [collection=img_deduplication] [partitions="[]"] [dsl=] [len(PlaceholderGroup)=1548] [OutputFields="[key]"] [search_params="[{\"key\":\"metric_type\",\"value\":\"IP\"},{\"key\":\"round_decimal\",\"value\":\"-1\"},{\"key\":\"anns_field\",\"value\":\"data\"},{\"key\":\"topk\",\"value\":\"10\"},{\"key\":\"params\",\"value\":\"{\\\"nprobe\\\":50}\"}]"] [travel_timestamp=0] [guarantee_timestamp=2]
[2022/09/13 17:39:40.697 +08:00] [DEBUG] [allocator/id.go:140] ["IDAllocator pickCanDoFunc"] [need=1] [total=197720] [remainReqCnt=0]
[2022/09/13 17:39:40.698 +08:00] [DEBUG] [timerecord/time_recorder.go:78] ["Search: search request enqueue (0ms)"]
[2022/09/13 17:39:40.698 +08:00] [DEBUG] [proxy/impl.go:2471] ["Search enqueued"] [traceID=3f3e3cd14dc83716] [role=proxy] [msgID=435961008929114345] [timestamp=435961719863640065] [db=] [collection=img_deduplication] [partitions="[]"] [dsl=] [len(PlaceholderGroup)=1548] [OutputFields="[key]"] [search_params="[{\"key\":\"metric_type\",\"value\":\"IP\"},{\"key\":\"round_decimal\",\"value\":\"-1\"},{\"key\":\"anns_field\",\"value\":\"data\"},{\"key\":\"topk\",\"value\":\"10\"},{\"key\":\"params\",\"value\":\"{\\\"nprobe\\\":50}\"}]"] [travel_timestamp=0] [guarantee_timestamp=2]
[2022/09/13 17:39:40.698 +08:00] [DEBUG] [proxy/task_search.go:217] ["translate output fields"] [msgID=435961008929114345] ["output fields"="[key]"]

**Comment by yanliang567** [comment]
sounds like a known issue that happens occasionally, let me check.

**Comment by ICLzhang** [comment]
etcd data
[bw_etcd_ALL.220913-232441.bak.gz](https://github.com/milvus-io/milvus/files/9563848/bw_etcd_ALL.220913-232441.bak.gz)

**Comment by yanliang567** [comment]
@ICLzhang @cjh6666688888 this issue should be fixed on milvusdb/milvus:2.1.0-20220913-3c3ba55, any chance to retry?

/assign @cjh6666688888 @ICLzhang

**Comment by ICLzhang** [comment]
ok，Service has been restored in milvus2.1.2。
